### PR TITLE
Add missing gRPC docs for the new features

### PR DESCRIPTION
**Pull request by alesj**

### CODE REVIEW

File: docs/src/main/asciidoc/grpc-getting-started.adoc [hunk 1]
```diff
@@ -178,6 +178,16 @@ quarkus.generate-code.grpc.scan-for-proto=<groupId>:<artifactId>
 ----
 The value of the property may be `none`, which is the default value, or a comma separated list of `groupId:artifactId` coordinates.
 
+== Different gRPC implementations / types
+
+Another thing to take note as well is that Quarkus' gRPC support currently includes 3 different types of gRPC usage:
+
+a. old Vert.x gRPC implementation with a separate gRPC server (default)
+b. new Vert.x gRPC implementation on top existing HTTP server
```
Comment (gsmet) [inline docs/src/main/asciidoc/grpc-getting-started.adoc:186]:
```suggestion
b. new Vert.x gRPC implementation on top of the existing HTTP server
```

File: docs/src/main/asciidoc/grpc-service-consumption.adoc [hunk 2]
```diff
@@ -209,6 +209,10 @@ The `client-name` is the name set in the `@GrpcClient` or derived from the injec
 The following examples uses _hello_ as the client name.
 Don't forget to replace it with the name you used in the `@GrpcClient` annotation.
 
+IMPORTANT: When you enable `quarkus.grpc.clients."client-name".use-quarkus-grpc-client`, you are then using new Vert.x gRPC channel implementation, so not all configuration properties can still be applied. And currently there is no `stork` support yet.
```
Comment (gsmet) [inline docs/src/main/asciidoc/grpc-service-consumption.adoc:212]:
```suggestion
IMPORTANT: When you enable `quarkus.grpc.clients."client-name".use-quarkus-grpc-client`, you are then using the new Vert.x gRPC channel implementation, so not all configuration properties can still be applied. And currently there is no Stork support yet.
```

File: docs/src/main/asciidoc/grpc-service-implementation.adoc [hunk 3]
```diff
@@ -212,6 +212,12 @@ If you wish to scale your server, you can set the number of server instances by
 
 include::{generated-dir}/config/quarkus-grpc-config-group-config-grpc-server-configuration.adoc[opts=optional, leveloffset=+1]
 
+IMPORTANT: When you disable `quarkus.grpc.server.use-separate-server`, you are then using new Vert.x gRPC server implementation
```
Comment (gsmet) [inline docs/src/main/asciidoc/grpc-service-implementation.adoc:215]:
```suggestion
IMPORTANT: When you disable `quarkus.grpc.server.use-separate-server`, you are then using the new Vert.x gRPC server implementation
```

File: docs/src/main/asciidoc/grpc-service-implementation.adoc [hunk 4]
```diff
@@ -212,6 +212,12 @@ If you wish to scale your server, you can set the number of server instances by
...
+which uses existing HTTP server. Which means that the server port is now `8080`. Also, most of the other configuration properties
```
Comment (gsmet) [inline docs/src/main/asciidoc/grpc-service-implementation.adoc:216]:
```suggestion
which uses the existing HTTP server. Which means that the server port is now `8080`. Also, most of the other configuration properties
```

File: docs/src/main/asciidoc/grpc-xds.adoc [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Using xDS gRPC
+include::_attributes.adoc[]
+:categories: serialization
+:summary: This page explains how to enable xDS gRPC usage in your Quarkus application.
+
+This page explains how to enable xDS gRPC usage in your Quarkus application.
+
+IMPORTANT: This Quarkus xDS gRPC integration currently doesn't support `native` runtime due to the issues
```
Comment (gsmet) [inline docs/src/main/asciidoc/grpc-xds.adoc:13]:
```suggestion
IMPORTANT: This Quarkus xDS gRPC integration currently doesn't support building native executables due to the issues
```

File: docs/src/main/asciidoc/grpc-xds.adoc [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+with sharded grpc-netty library while running native IT tests.
```
Comment (gsmet) [inline docs/src/main/asciidoc/grpc-xds.adoc:14]: Either `shaded` or `shared`?
```suggestion
with shaded grpc-netty library while running native IT tests.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
I added a few wording suggestions. I will let @cescoffier check the content.

File: docs/src/main/asciidoc/grpc-getting-started.adoc [hunk 7]
```diff
@@ -178,6 +178,16 @@ quarkus.generate-code.grpc.scan-for-proto=<groupId>:<artifactId>
...
+c. xDS wrapper over https://github.com/grpc/grpc-java[grpc-java] with a separate Netty based gRPC server
```
Comment (cescoffier) [inline docs/src/main/asciidoc/grpc-getting-started.adoc:187]: A link explaining what xDS is would be useful.

File: docs/src/main/asciidoc/grpc-service-consumption.adoc [hunk 8]
```diff
@@ -401,6 +405,47 @@ To disable the gRPC client metrics when `quarkus-micrometer` is used, add the fo
...
+@ApplicationScoped
+public class HelloExceptionHandlerProvider implements ExceptionHandlerProvider {
+    @Override
+    public <ReqT, RespT> ExceptionHandler<ReqT, RespT> createHandler(ServerCall.Listener<ReqT> listener,
+            ServerCall<ReqT, RespT> serverCall, Metadata metadata) {
+        return new HelloExceptionHandler<>(listener, serverCall, metadata);
+    }
+
+    @Override
+    public Throwable transform(Throwable t) {
```
Comment (cescoffier) [inline docs/src/main/asciidoc/grpc-service-consumption.adoc:426]: Should the returned type be a StatusRuntime Exception?

File: docs/src/main/asciidoc/grpc-service-implementation.adoc (hunk 4 above)
Comment (cescoffier) [inline docs/src/main/asciidoc/grpc-service-implementation.adoc:216]:
```suggestion
which uses existing HTTP server. Which means that the server port is now `8080` (or the port configured with `quarkus.http.port`). Also, most of the other configuration properties
```

File: docs/src/main/asciidoc/grpc-xds.adoc [hunk 9]
```diff
@@ -0,0 +1,83 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Using xDS gRPC
```
Comment (cescoffier) [inline docs/src/main/asciidoc/grpc-xds.adoc:6]: This heading will make Vale very unhappy :-D 
We need to add `xDS` and `gRPC` to the excluded word list (@michelle-purcell / @ebullient )

File: docs/src/main/asciidoc/grpc-xds.adoc (hunk 6 above)
Comment (cescoffier) [inline docs/src/main/asciidoc/grpc-xds.adoc:14]:
```suggestion
with shaded `grpc-netty` library while running native IT tests.
```

Comment (cescoffier) [reply docs/src/main/asciidoc/grpc-xds.adoc:14]: shaded :-D 

But I think we can write a _law_: all libraries using a shaded netty variant are sharding the netty space

File: docs/src/main/asciidoc/grpc-xds.adoc [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-grpc-xds</artifactId>
+</dependency>
+----
+
+NOTE: This transitively adds `io.quarkus:quarkus-grpc` extension dependency.
+
+== Server Configuration
```
Comment (cescoffier) [inline docs/src/main/asciidoc/grpc-xds.adoc:30]:
```suggestion
== Server configuration
```

File: docs/src/main/asciidoc/grpc-xds.adoc [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+include::{generated-dir}/config/quarkus-grpc-config-group-config-xds.adoc[opts=optional, leveloffset=+1]
+
+== Example of Server Configuration
```
Comment (cescoffier) [inline docs/src/main/asciidoc/grpc-xds.adoc:34]:
```suggestion
== Server configuration example
```

File: docs/src/main/asciidoc/grpc-xds.adoc [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+To enable server xDS, use the following configuration.
+
+xDS must be explicitly enabled for server, and make sure you point it to the right xDS server port (default is 9000).
```
Comment (cescoffier) [inline docs/src/main/asciidoc/grpc-xds.adoc:38]:
```suggestion
xDS must be explicitly enabled on the server,  then verify you use it to the right xDS server port (default is 9000).
```

File: docs/src/main/asciidoc/grpc-xds.adoc [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+[source,properties]
+----
+quarkus.grpc.server.xds.enabled=true
+#quarkus.grpc.server.xds.secure=true
+quarkus.grpc.server.port=30051
+----
+
+NOTE: When xDS is configured, `plain-text` is automatically disabled.
+
+== Client Configuration
```
Comment (cescoffier) [inline docs/src/main/asciidoc/grpc-xds.adoc:50]:
```suggestion
== Client configuration
```

File: docs/src/main/asciidoc/grpc-xds.adoc [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+
+include::{generated-dir}/config/quarkus-grpc-config-group-config-client-xds.adoc[opts=optional, leveloffset=+1]
+
+NOTE: When xDS target property is used, name resolver, host and port are not used
```
Comment (cescoffier) [inline docs/src/main/asciidoc/grpc-xds.adoc:54]:
```suggestion
NOTE: When xDS target property is used, name resolver, host, and port are not used.
```

File: docs/src/main/asciidoc/grpc-xds.adoc [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+
+== Example of Client Configuration
```
Comment (cescoffier) [inline docs/src/main/asciidoc/grpc-xds.adoc:56]:
```suggestion
== Client configuration example
```

File: docs/src/main/asciidoc/grpc-xds.adoc [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+----
+#quarkus.grpc.clients.<client-name>.xds.enabled=true
+#quarkus.grpc.clients.<client-name>.xds.secure=true
+quarkus.grpc.clients.<client-name>.name-resolver=xds
+quarkus.grpc.clients.<client-name>.port=30051
+----
+
+NOTE: When xDS is configured, `plain-text` is automatically disabled.
+
+== Example of additional Kubernetes Configuration
```
Comment (cescoffier) [inline docs/src/main/asciidoc/grpc-xds.adoc:74]:
```suggestion
== Kubernetes configuration example
```

File: docs/src/main/asciidoc/grpc-xds.adoc [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+
+Below is an example of (required) additional configuration when using xDS gRPC with Istio in Kubernetes.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/grpc-xds.adoc:76]:
```suggestion
Below is an example of (required) additional configuration when using xDS gRPC with the Istio Service Mesh in Kubernetes.
```

File: docs/src/main/asciidoc/grpc-getting-started.adoc (hunk 7 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/grpc-getting-started.adoc:187]: something like https://grpc.github.io/grpc/core/md_doc_grpc_xds_features.html[xDS gRPC]

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/grpc-service-consumption.adoc (hunk 8 above)
Comment (alesj) [reply docs/src/main/asciidoc/grpc-service-consumption.adoc:426]: No need + there is a problem of which one to return -- StatusRuntimeException or plain StatusException ...
I guess users will mostly return one of them ... but they can opt into other runtime exceptoins as well, if they want them to propagate up the stack ...

File: docs/src/main/asciidoc/grpc-xds.adoc (hunk 9 above)
Comment (michelle-purcell) [reply docs/src/main/asciidoc/grpc-xds.adoc:6]: Fixed in PR #30256.

**Pull request merged**